unittests/allplatformstests.py

```diff
@@ -2164,7 +2164,7 @@ def test_templates(self):
 
         langs = ['c']
         env = get_fake_env()
-        for l in ['cpp', 'cs', 'd', 'java', 'cuda', 'fortran', 'objc', 'objcpp', 'rust']:
+        for l in ['cpp', 'cs', 'd', 'java', 'cuda', 'fortran', 'objc', 'objcpp', 'rust', 'vala']:
```

Comment:
Does it make sense to use a set here while we are at it?

Reply:
No, set membership tests are really fast but there's no advantage for iteration. In fact, sets are slower at iteration.

Reply:
On the other hand, this does not need to be a list:
```
python
for l in 'cpp', 'cs', 'd', 'java', 'cuda', 'fortran', 'objc', 'objcpp', 'rust', 'vala':
    pass
```
so you spare the few microseconds needed to instantiate a list object :)

Reply:
That instantiates a tuple object instead, because the `('x', 'y')` parentheses are optional in tuple declaration if the context is unambiguous.

Tuple vs list doesn't really matter in this context, as they essentially cost exactly the same, although I do happen to think the implicit parentheses makes for surprising reading. and would prefer to avoid it.